skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -128,4 +134,37 @@ public EndorsementModel createEndorsement(EndorsementDRO endorsementDRO) {
             throw new NoEntityException("Skill with id:" + skillId + " not found");
         }
     }
+
+    @Override
+    public GenericResponse<Void> updateEndorsementStatus(String id, String status) {
```

Comment:
1.  Currently, the function doesn't handle potential race conditions if multiple users try to update the same endorsement simultaneously. To address this, you could implement optimistic locking using a version field in the EndorsementModel. This would require additional logic to handle potential conflicts during saving.
2. The code checks if the status is either APPROVED or REJECTED using name(). This might be redundant depending on the implementation of EndorsementStatus. If EndorsementStatus is an enum with values APPROVED and REJECTED, then the check can be simplified to

Reply:
can we connect regarding the 1st point?
And at the end of 2nd point I think you were suggesting a better approach, can you complete it?

Reply:
Ok, are you suggesting me to add a property in the EndrorsementModel which signifies whether the endorsement  status is already updated?
Such that, whenever a model with status as APPROVED/REJECTED is again being patched should be avoided?

Do let me know if I got it wrong

Reply:
pushed new changes